addons/binding/org.openhab.binding.foobot/src/main/java/org/openhab/binding/foobot/handler/FoobotHandler.java

```diff
@@ -0,0 +1,284 @@
+                updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.CONFIGURATION_ERROR, errorMsg);
+
+                return;
+            }
+            // Map the JSON response to list of objects
+            Type listType = new TypeToken<ArrayList<FoobotJsonResponse>>() {
+            }.getType();
+            String userDevices = EntityUtils.toString(response.getEntity());
+            List<FoobotJsonResponse> readFromJson = gson.fromJson(userDevices, listType);
+            for (FoobotJsonResponse ob : readFromJson) {
+                // Compare the mac address to each record in order to fetch the UUID of the current device.
+                if (ob.getMac().equals(config.mac)) {
+                    uuid = ob.getUuid();
+                    deviceFound = true;
+                    break;
+                }
+            }
+            if (!deviceFound) {
+                errorMsg = "No device found for this MAC address.";
+                logger.warn("Error in foobot.io response: {}", errorMsg);
+                updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.CONFIGURATION_ERROR, errorMsg);
+
+                return;
+            }
+        } catch (MalformedURLException e) {
+            logger.warn("Constructed url is not valid: {}", e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+        } catch (JsonSyntaxException e) {
+            logger.warn("Error running foobot.io request: {}", e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+        } catch (IOException | IllegalStateException e) {
+            logger.warn("Error running foobot.io request: {}", e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+        }
+        updateStatus(ThingStatus.ONLINE);
+        startAutomaticRefresh();
+    }
+
+    /**
+     * Start the job refreshing the Sensor data from Foobot device
+     */
+    private void startAutomaticRefresh() {
+        int delay = (config.refresh != null) ? config.refresh.intValue() : DEFAULT_REFRESH_PERIOD;
+        refreshJob = scheduler.scheduleWithFixedDelay(() -> {
+            // Request new sensor data from foobot.io service
+            foobotData = updateFoobotData();
+
+            // Update all channels from the updated Sensor data
+            for (Channel channel : getThing().getChannels()) {
+                updateChannel(channel.getUID().getId(), foobotData);
+            }
+        }, 0, delay, TimeUnit.MINUTES);
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing the Foobot handler.");
+        if (refreshJob != null && !refreshJob.isCancelled()) {
+            refreshJob.cancel(true);
+            refreshJob = null;
+        }
+    }
+
+    @Override
+    public void handleCommand(@NonNull ChannelUID channelUID, @NonNull Command command) {
```

Comment:
When adding null annotations, please only use the `@NonNullByDefault` and `@Nullable` annotations. See also the [Null Annotations Conventions](https://www.openhab.org/docs/developer/development/conventions.html#null-annotations).